File: scripts/clean_search_index.py
import json
import os

# Define relative path from the script's intended execution context (project root)
file_path = os.path.join('data', 'search_index.json')

try:
    if not os.path.exists(file_path):
        print(f"Error: {file_path} not found.")
        exit(1)

    with open(file_path, 'r', encoding='utf-8') as f:
        data = json.load(f)

    print(f"Loaded {len(data)} items. Removing 'folder' keys...")

    # Remove 'folder' field from each dictionary
    for item in data:
        if 'folder' in item:
            del item['folder']

    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=2)

    print("Successfully cleaned search_index.json.")
except Exception as e:
    print(f"Error: {str(e)}")
    exit(1)

File: scripts/clean_search_index.js
const fs = require('fs');
const path = require('path');

// Use relative path from the project root (web/data/search_index.json)
const filePath = path.join(__dirname, '..', 'data', 'search_index.json');

try {
    if (!fs.existsSync(filePath)) {
        console.error(`File not found: ${filePath}`);
        process.exit(1);
    }

    const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
    console.log(`Loaded ${data.length} items. Removing "folder" metadata...`);

    const cleanedData = data.map(item => {
        const { folder, ...rest } = item;
        return rest;
    });

    fs.writeFileSync(filePath, JSON.stringify(cleanedData, null, 2));
    console.log('Successfully cleaned search_index.json and saved relative to project root.');
} catch (error) {
    console.error('Error processing search_index.json:', error.message);
    process.exit(1);
}
